editor/editor_node.cpp

```diff
@@ -983,7 +986,10 @@ void EditorNode::_dialog_action(String p_file) {
 			if (file->get_mode() == EditorFileDialog::MODE_SAVE_FILE) {
 
 				_save_default_environment();
-				_save_scene_with_preview(p_file);
+				if (scene_idx != editor_data.get_edited_scene() || _get_current_main_editor() == EDITOR_SCRIPT)
```

Comment:
yeah i totally did not realize the first condition.
Fix is right, but just leave the condition as:
scene_idx != editor_data.get_edited_scene() 

as checking whether you are in spatial editor or not does not make any sense now